### PR TITLE
feat(resources): add kuma.io/display-name label

### LINKED ISSUE #8484: Rule merging alghoritm is not sorting lexographically on MeshTrafficPermission
### What happened?

My intention is to create a `MeshTrafficPermission` set of rules that will allow connectivity from some services and ShadowDeny (or Deny) everything else. 

To achieve that i did a 2 `MeshTrafficPermission` objects, both on  `MeshSubset` targetRef level.

```yaml
apiVersion: kuma.io/v1alpha1
kind: MeshTrafficPermission
metadata:
  labels:
    kuma.io/mesh: default
  name: default-demo-app
  namespace: kuma-system
spec:
  targetRef:
    kind: MeshSubset
    tags:
      k8s.kuma.io/service-name: demo-app
  from:
    - default:
        action: Allow
      targetRef:
        kind: MeshSubset
        tags:
          app.kubernetes.io/name: ui
    - default:
        action: Allow
      targetRef:
        kind: MeshSubset
        tags:
          app.kubernetes.io/name: service-order
    - default:
        action: Allow
      targetRef:
        kind: MeshSubset
        tags:
          app.kubernetes.io/name: service-payment
---

apiVersion: kuma.io/v1alpha1
kind: MeshTrafficPermission
metadata:
  labels:
    kuma.io/mesh: default
  name: default-demo-app-b
  namespace: kuma-system
spec:
  targetRef:
    kind: MeshSubset
    tags:
      k8s.kuma.io/service-name: demo-app
  from:
    - default:
        action: Deny
      targetRef:
        kind: Mesh
```

This setup works fine if naming is like `myrule` for allow set and `myrule-a` for denyset. I see in UI that rules are applied in correct order:
Screenshot 2023-11-28 at 15 54 03

My tests also shows that traffic works as expected

However, when I try to use names like `rule-a` and `rule-b` order is opposite to expected and tests also failing.

Screenshot 2023-11-28 at 15 56 25

I am not sure how sorting is working, but looks like that not according to doc which state that `2. At the same level we use lexicographic order on name`.

### PR DESCRIPTION
**Pull request by jakubdyszkiewicz**
### Checklist prior to review

Fix #8484

This PR adds a new standard label `kuma.io/display-name`.

`kuma.io/display-name` is needed for the following cases:
* To preserve original Kubernetes name without namespace
* To preserve original name after KDS sync, because we add hashes now.
* To use it in matching to fix issue linked above.

We could add always it (postgres, memory store), but it's a bit awkward to repeat `name` field when it's equal.

If needed, I can write MADR for this change, but I'm not sure it's worth it.

- [X] [Link to relevant issue][1] as well as docs and UI issues --
- [X] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [X] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [X] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [X] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: api/mesh/v1alpha1/dataplane_helpers.go [hunk 1]
```diff
@@ -42,6 +42,11 @@ const (
 
 	// Used for Service-less dataplanes
 	TCPPortReserved = 49151 // IANA Reserved
+
+	// DisplayName is a standard label that can be used to easier recognize policy name.
+	// On Kubernetes, Kuma resource name contains namespace. Display name is original name without namespace.
+	// The name contains hash when the resource is synced from global to sync. In this case, display name is original name from originated CP.
```
Comment (lobkovilya) [inline api/mesh/v1alpha1/dataplane_helpers.go:48]: typo
```suggestion
	// The name contains hash when the resource is synced from global to zone. In this case, display name is original name from originated CP.
```

**Review by lobkovilya — APPROVED** [review]

**Review by lahabana — APPROVED** [review]
Hmm wouldn't it just make sense to use it always? After all how long would it be until we do things differently in memory and zones?

**Comment by jakubdyszkiewicz** [comment]
by "use it always" do you mean add it always to every single store?

**Comment by lahabana** [comment]
> by "use it always" do you mean add it always to every single store?

Yeah but I could be wrong...

**Comment by jakubdyszkiewicz** [comment]
We could always add it if we find this to be problematic. It's hard to go the other way around

**Pull request merged**